duckdb-vortex/src/vortex_extension.cpp

```diff
@@ -47,51 +52,52 @@ struct VortexBindData : public TableFunctionData {
 struct VortexScanState : public LocalTableFunctionState {
 	idx_t current_row = 0;
 	bool finished = false;
+	mutable Array *array;
+
+	optional_ptr<TableFilterSet> filter;
+	vector<idx_t> column_ids;
 };
 
 static void VortexScanFunction(ClientContext &context, TableFunctionInput &data, DataChunk &output) {
 	auto &bind_data = data.bind_data->Cast<VortexBindData>(); // NOLINT
-	auto &state = data.local_state->Cast<VortexScanState>();
+	auto &state = data.local_state->Cast<VortexScanState>();  // NOLINT
+
+	if (bind_data.array_stream == nullptr) {
+		const char **const column_names = new const char *[state.column_ids.size()];
+		auto idx = 0;
+		for (auto col_id : state.column_ids) {
```

Comment:
Leaning quite far into c-style coding here. Guess that's we gotta do.

Reply:
any suggestions would be great

Reply:
Not sure I guess we need this due to `FileScanOptions`.